bot/cogs/moderation/silence.py

```diff
@@ -0,0 +1,175 @@
+        """
+        log.debug(f"{ctx.author} is silencing channel #{ctx.channel}.")
+        if not await self._silence(ctx.channel, persistent=(duration is None), duration=duration):
+            await ctx.send(f"{Emojis.cross_mark} current channel is already silenced.")
+            return
+        if duration is None:
+            await ctx.send(f"{Emojis.check_mark} silenced current channel indefinitely.")
+            return
+
+        await ctx.send(f"{Emojis.check_mark} silenced current channel for {duration} minute(s).")
+        await asyncio.sleep(duration*60)
+        log.info(f"Unsilencing channel after set delay.")
+        await ctx.invoke(self.unsilence)
+
+    @commands.command(aliases=("unhush",))
+    async def unsilence(self, ctx: Context) -> None:
+        """
+        Unsilence `channel`.
+
+        Unsilence a previously silenced `channel`,
+        remove it from notifier of indefinitely silenced channels and cancel the notifier if empty.
+        """
+        log.debug(f"Unsilencing channel #{ctx.channel} from {ctx.author}'s command.")
+        if await self._unsilence(ctx.channel):
+            await ctx.send(f"{Emojis.check_mark} unsilenced current channel.")
+
+    async def _silence(self, channel: TextChannel, persistent: bool, duration: Optional[int]) -> bool:
+        """
+        Silence `channel` for `self._verified_role`.
+
+        If `persistent` is `True` add `channel` to notifier.
+        `duration` is only used for logging; if None is passed `persistent` should be True to not log None.
+        """
+        current_overwrite = channel.overwrites_for(self._verified_role)
+        if current_overwrite.send_messages is False:
+            log.info(f"Tried to silence channel #{channel} ({channel.id}) but the channel was already silenced.")
+            return False
+        await channel.set_permissions(
+            self._verified_role,
+            overwrite=PermissionOverwrite(**dict(current_overwrite, send_messages=False))
+        )
+        self.muted_channels.add(channel)
+        if persistent:
+            log.info(f"Silenced #{channel} ({channel.id}) indefinitely.")
+            self.notifier.add_channel(channel)
+            return True
+
+        log.info(f"Silenced #{channel} ({channel.id}) for {duration} minute(s).")
+        return True
+
+    async def _unsilence(self, channel: TextChannel) -> bool:
+        """
+        Unsilence `channel`.
+
+        Check if `channel` is silenced through a `PermissionOverwrite`,
+        if it is unsilence it and remove it from the notifier.
+        """
+        current_overwrite = channel.overwrites_for(self._verified_role)
+        if current_overwrite.send_messages is False:
+            await channel.set_permissions(
+                self._verified_role,
+                overwrite=PermissionOverwrite(**dict(current_overwrite, send_messages=True))
```

Comment:
As mentioned before, permissions can be passed as kwargs rather than using an entire `PermissionOverwrite` object.